pkg/codegen/docs_test.go

```diff
@@ -119,3 +119,251 @@ func fakeFunc() {
 			"unexpected Example 2 section. section should have been excluded")
 	})
 }
+
+func TestInterpretPulumiRefs(t *testing.T) {
+	t.Parallel()
+
+	t.Run("ResolvesKnownRefs", func(t *testing.T) {
+		t.Parallel()
+
+		description := "This is a reference to <pulumi ref=\"#/resources/aws:s3:bucket\" /> and one to the " +
+			"<pulumi ref=\"#/resources/azure:storage:storageAccount/properties/region\" /> property."
+		expected := "This is a reference to s3.bucket and one to the region property."
+		result := InterpretPulumiRefs(description, func(ref DocRef) (string, bool) {
+			//nolint:exhaustive
+			switch ref.Type {
+			case DocRefTypeResource:
+				return ref.Token.Module().Name().String() + "." + ref.Token.Name().String(), true
+			default:
+				return "", false
+			}
+		})
+		assert.Equal(t, expected, result, "expected resolved references")
+	})
+
+	t.Run("FallsBackWhenNotMapped", func(t *testing.T) {
+		t.Parallel()
+
+		description := "This is a reference to <pulumi ref=\"#/resources/unknown:resource:type/properties/myProperty\"/>" +
+			" and to <pulumi ref=\"#/resources/unknown:resource:type\" />."
+		expected := "This is a reference to myProperty and to unknown:resource:type."
```

Comment:
Should this be unknown:resource:type.property? To scope it

Reply:
Maybe, but that's not as nice looking .. if we had the context of which resource we're generating docs for, then perhaps we can differentiate between other resource's properties and the current resource's property.

Reply:
Yeh leave it as is, simple fallback is fine.